fix(http): Include HTTP status code and headers when HTTP requests errored in `httpResource`

wartab opened the pull request:
Currently the HTTP status code and headers are only included if the request succeeded. Given status codes convey more information in case of a request error vs. success, this makes it more useful than inspecting what is contained in `.error()`.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information
I'm not sure if the current behaviour is actually intended, but I don't really see why statusCode would exist if it was not meant to contain something else than 200 (or 204, but you get my point).

JeanMeche reviewed (approved):
It looks like it broke some tests, can you have a look at it. Thank you.

wartab commented:
Yeah, just saw it, I tried to run them on my computer, but there were a lot of errors that were related to not being able to run the chrome sandboxes that crashed with SEGFAULTs.

JeanMeche commented:
You should be able to run the node only test with `yarn bazel //packages/common/http/test:test`

wartab committed:
fix(http): Include HTTP status code and headers when HTTP requests errored in `httpResource`

Currently the HTTP status code and headers are only included if the request succeeded. Given status codes convey more information in case of a request error vs. success, this makes it more useful than inspecting what is contained in `.error()`.

wartab commented:
Ah thanks, that is faster as well :') I ran it on //packages/common

atscott commented:
This PR was merged into the repository by commit 9f31947aad8331a02166d87d98ac2d4ed5aa701b.

The changes were merged into the following branches: main, 19.2.x
